lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -192,6 +192,61 @@ defmodule Mix.Tasks.Xref do
     :ok
   end
 
+  @doc """
+  Returns a list of information of all the function calls in the project.
+
+  Each item in the list is a map with the following keys:
+
+    * `:callee` - A tuple containing the module, function and arity of the call
+    * `:line` - Line where the function is called (integer)
+    * `:file` - File where the function is called (binary)
+    * `:caller_module` - Module where the function is called (atom)
+
+  This function returns an empty list when used at the root of an umbrella
+  project because there is no compile manifest to extract the function call
+  information from. To get the function calls of each child in an umbrella,
+  execute the function at the root of each individual application.
+  """
+  @spec calls() :: [map]
```

Comment:
Can you please update the spec to be something like:

  @spec calls() :: [%{callee: {module, atom, arity}, caller_module: module, line: integer(), file: String.t]

Please don't forget to run the formatter after you do. :) Thanks!

Reply:
Ok no problem!